userspace/userspace-common/src/shared_object_cache.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (C) 2019 Sysdig Inc.
+
+This file is part of sysdig.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <shared_object_cache.h>
+
+namespace userspace_common
+{
+
+template<class TKey, class TValue>
+void shared_object_cache<TKey, TValue>::insert(const TKey& key, const value_ptr_t& value)
```

Comment:
This behaves different from unordered_map::insert when the element already exists (overwrites it while unordered_map::insert doesn't). Maybe rename this to `set`, `put` or `insert_or_assign` a'la C++17?